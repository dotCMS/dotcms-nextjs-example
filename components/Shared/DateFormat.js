const DateFormat = ({ value }) => {
    const monthNames = [
        'January',
        'February',
        'March',
        'April',
        'May',
        'June',
        'July',
        'August',
        'September',
        'October',
        'November',
        'December'
    ];
    const d = new Date(value);

    return <time dateTime={value}>{`${monthNames[d.getMonth()]} ${d.getUTCDate()}, ${d.getFullYear()}`}</time>;
};

export default DateFormat;
