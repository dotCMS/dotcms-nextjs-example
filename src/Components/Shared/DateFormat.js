import React from 'react';

const DateFormat = ({dateString}) => {
    const options = { year: 'numeric', month: 'long', day: 'numeric' };
    const dateText =  new Date(dateString).toLocaleDateString('en-US', options);
    return <time dateTime={dateString}>{dateText}</time>;
};

export default DateFormat;
