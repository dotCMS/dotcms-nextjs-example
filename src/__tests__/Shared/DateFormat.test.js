import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import DateFormat from '../../Components/Shared/DateFormat';

describe('<DateFormat />', () => {
    it('renders and matches snapshot', () => {
        const wrapper = shallow(<DateFormat dateString="2016-01-25 07:47:00.0" />);
        expect(toJSON(wrapper)).toMatchSnapshot();
    });
});
