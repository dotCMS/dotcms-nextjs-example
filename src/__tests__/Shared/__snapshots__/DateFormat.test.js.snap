// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DateFormat /> renders and matches snapshot 1`] = `
<time
  dateTime="2016-01-25 07:47:00.0"
>
  January 25, 2016
</time>
`;
